### PR TITLE
Add API compatibility level task

### PR DESCRIPTION
**Pull request by Azurelol**
## Description

Adds a new task to set the API compatibility level on the ProjectSettings.asset, which will be used to test Unity versions against different NET runtimes (2, 4.6)

## Changes
* ![ADD] SetAPICompatibilityLevel
* ![ADD] APICompatibilityLevel
* ![UPDATE] DefaultUnityPluginExtension, UnityPlugin
* ![UPDATE] UnityPluginConsts, UnityPluginConvention

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+//        def result = runTasksWithFailure("test")
+//
+//        then:
+//        // check for specific error message
+//        result.standardError.contains(SetAPICompatibilityLevel.parseFailureMessage)
+//
+//        where:
+//        defaultAPICompatibilityLevel = APICompatibilityLevel.net2_0_subset
+//        expectedAPICompatibilityLevel << [5, "lol"]
+//    }
```
Comment (Larusso) [inline src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy:166]: ? why is this commented out?

File: src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy [hunk 2]
```diff
@@ -55,40 +57,22 @@ import java.util.concurrent.Callable
...
+ *    exportUnityPackage {*        inputFiles file('Assets')
+ *}*
+ *    task(performBuild, type:wooga.gradle.unity.tasks.Unity) {*        args "-executeMethod", "MyEditorScript.PerformBuild"
+ *}*
+ *    task(performMultipleBuilds) {*        doLast {*            unity.batchMode {*                unityPath = project.file("/Applications/Unity-5.5.3f1/Unity.app/Contents/MacOS/Unity")
+ *                args "-executeMethod", "MyEditorScript.PerformBuild"
+ *}*
+ *            unity.batchMode {*                unityPath = project.file("/Applications/Unity-5.6.0f3/Unity.app/Contents/MacOS/Unity")
+ *                args "-executeMethod", "MyEditorScript.PerformBuild"
+ *}*}*}*}
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy:75]: can you repair the example?

**Review by Larusso — CHANGES_REQUESTED** [review]

File: src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy (hunk 2 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy:75]: of course

File: src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy (hunk 1 above)
Comment (Azurelol) [reply src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy:166]: This was one of the earlier tests I tried to write but wasn't sure if it would reflect the intended behaviour.
Since this was more about testing how API comp level would be parsed, I tested it there (for integers, as strings and null will throw).
If an invalid type is passed for the argument it will throw since the signature won't match. Is this something we handle with overloads or is that unneeded?

Comment (Azurelol) [reply src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy:166]: As in write a new override on DefaultUnityPLuginExtension for type Object and handle this there?

Comment (Larusso) [reply src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy]: I think if you handle the parsing in the Unittest for the enum it should be enough.

**Review by Larusso — APPROVED** [review]

File: src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy (hunk 1 above)
Comment (Azurelol) [reply src/integrationTest/groovy/wooga/gradle/unity/SetAPICompLevelIntegrationSpec.groovy]: Very well.
